### PR TITLE
Fixed issue where merging objects constructed from custom class was causing unexpected result  

### PR DESCRIPTION
**Pull request by sumeetkakkar**
Don't merge objects created from custom Classes. The merge can fail or return unexpected results depending on the Class implementation.

Below are additional changes done under this PR:
- Fixed case where both initial promise and resolveImport promise in Factory._add were rejected and this caused UnhandledPromiseRejectionWarning causing unit test to fail.
- Updated dependencies
- Removed babel dependency for compiling

### CODE REVIEW

File: lib/common.js [hunk 1]
```diff
@@ -35,8 +37,8 @@ export default {
     },
 
     merge(src, dest) {
-        // NOTE: Do not merge arrays and only merge objects into objects.
-        if (!Thing.isArray(src) && Thing.isObject(src) && Thing.isObject(dest)) {
+        // NOTE: Do not merge arrays and only merge objects into objects. Do not merge special objects.
```
Comment (kumarrishav) [inline lib/common.js]: What does it mean by special objects? can we clarify it more (with a bit of reasoning probably)? I can see there is check of getPrototypeof

**Comment by kumarrishav** [comment]
PR name is incomplete I think. as it does more than that (that prototype check )
. Something like
`Remove babel dependency, Merge only enumerable objects (if that's what it's doing? )`

Remove babel dependency. => means no need for compiling. So, I think no need to mention explicitly

File: lib/common.js (hunk 1 above)
Comment (sumeetkakkar) [reply lib/common.js]: Updated the comment

File: .travis.yml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 language: node_js
 node_js:
-  - "4"
-  - "6"
   - "8"
```
Comment (kumarrishav) [inline .travis.yml:3]: drop 8?

File: package.json [hunk 3]
```diff
@@ -31,21 +28,19 @@
   ],
   "readmeFilename": "README.md",
   "devDependencies": {
-    "babel": "4.7.3",
-    "glob": "^4.3.5",
-    "jshint": "^2.6.0",
-    "nyc": "^3.0.1",
-    "tap": "^1.3.1"
+    "glob": "^7.1.6",
+    "jshint": "^2.11.0",
```
Comment (kumarrishav) [inline package.json:32]: are we using linting anywhere?

Comment (sumeetkakkar) [reply package.json:32]: It seems this dependency was not removed when linting was removed. https://github.com/krakenjs/confit/commit/39a3963f2444018fae3d0d1607afb9eb8a84ce21. Maybe for the IDE?

File: .travis.yml (hunk 2 above)
Comment (sumeetkakkar) [reply .travis.yml:3]: Maybe next time when we add 14. Decided to keep three supported versions.

**Review by kumarrishav — APPROVED** [review]

**Review by grawk — APPROVED** [review]
Looks good to me.

**Pull request merged**